### ISSUE TITLE
Fetching data using accesible_by

### ISSUE THREAD

**Issue by spatil** [ryanb/cancan#883]
Relations : 
Order belongs_to user
Order has_many line_item_approvers
Company has_many users

I have defined the abilities like this : 

can :manage, Order, user_id: @resource.companies.collect(&:user_ids)
cannot :manage, Order, status: 'pending'
can :manage, Order, user_id: @user.id
can :manage, Order, line_item_approvers: {user_id: @user.id}

But When I do Order.accessible_by it always give me the results from this query 
can :manage, Order, line_item_approvers: {user_id: @user.id}

Basically I want to have something like this 

User can manage his own order, cancan manage orders except pending once & 
( can manage his own company orders (any user order which belongs to his company) 
[ can :manage, Order, user_id: @resource.companies.collect(&:user_ids) ]
OR
Orders items approver include this user ) 
[ can :manage, Order, line_item_approvers: {user_id: @user.id} ]

Any help ?
